BREAKING CHANGE detected `@storybook/addon-interactions@8.2.0` removes `@storybook/test` module

### Discussed in https://github.com/storybookjs/storybook/discussions/28516

<div type='discussions-op-text'>

<sup>Originally posted by **valpioner** July 10, 2024</sup>
### Summary

Updating storybook to 8.2.0 broke our project.

Context:
We use `import { fn } from '@storybook/test';`
After we updated to 8.2.0, this module is not there any more.
After additional investigation, looks like `@storybook/addon-interactions@8.2.0` is responsible for that.

TS Error:
`Cannot find module '@storybook/test' or its corresponding type declarations.ts(2307)`

<img width="927" alt="image" src="https://github.com/storybookjs/storybook/assets/12964569/2e7e8162-03e5-4529-9296-50e2c07b2898">


### Additional information

_No response_

### Create a reproduction

_No response_</div>

--- a/code/addons/interactions/package.json
+++ b/code/addons/interactions/package.json
@@ -55,6 +55,7 @@
   "dependencies": {
     "@storybook/global": "^5.0.0",
     "@storybook/instrumenter": "workspace:*",
+    "@storybook/test": "workspace:*",
     "polished": "^4.2.2",
     "ts-dedent": "^2.2.0"
   },

